### PR TITLE
Jacob delete post

### PR DESCRIPTION
**Pull request by Jet004**
This pull request creates a route for deleting posts with cascade deletion of related postTags and likes. Comment deletion will need to be handled separately as it will require a custom recursive query to delete all comments AND related replies

### CODE REVIEW

**Review by QiujuYang — APPROVED** [review]

File: __test__/integrationTests.test.ts [hunk 1]
```diff
@@ -743,6 +743,105 @@ describe("Integration tests for POST routes:", () => {
...
+                status: "DRAFT",
+                category: "GENERAL",
+                postTags: ["JS"],
+            });
+            expect(await db.post.count()).toBe(1);
+
+            // Delete the post
+            const response = await axios({
+                url: "/api/posts/deletePost",
+                method: "POST",
```
Comment (2016lisali) [inline __test__/integrationTests.test.ts]: Is it ok to use method "post" instead of "delete" here?

File: __test__/integrationTests.test.ts [hunk 2]
```diff
@@ -743,6 +743,105 @@ describe("Integration tests for POST routes:", () => {
...
+            });
+            expect(loginResponse.status).toBe(200);
+
+            // Get session cookie
+            if (!loginResponse.headers["set-cookie"])
+                throw new Error("No cookie set");
+            const cookie: string = loginResponse.headers["set-cookie"][0];
+
+            // Delete the post
+            const response = await axios({
```
Comment (2016lisali) [inline __test__/integrationTests.test.ts:822]: Same question as above

File: routes/postRoutes.ts [hunk 3]
```diff
@@ -27,4 +28,11 @@ router
         updatePost(db.post.getPostById, db.post.updatePost)
     );
 
+router
+    .route("/deletePost")
+    .post(
```
Comment (2016lisali) [inline routes/postRoutes.ts]: Same question about using 'post' instead of 'delete'

**Review by galarist — APPROVED** [review]

**Review by 2016lisali — APPROVED** [review]
Great work👍

**Pull request merged**